### PR TITLE
Parametrize values

### LINKED ISSUE #2: Missing Readme file
Readme file is missing. 

The file should include: 
- What is this repo? 
- Why is this repo? (It's purpose)
- How to use it. (At least instructions for a quick start)

### PR DESCRIPTION
**Pull request by estradag**
Closes #2 

- Modified the Ansible part of the project to give it the structure of an Ansible role. It will be taken out to its own repo in a future update.
- Parametrized static page repository and the bucket/domain that should be used to access the page once it has been uploaded to the corresponding S3 bucket.
- `www` subdomain S3 bucket and Route53 record are optional.

### CODE REVIEW

**Review by cracos — APPROVED** [review]
LTMG, don't forget in the future apply enhancements

File: ansible-s3-static-page-upload-role/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+repo: git@github.com:Tinker-Ware/homepage.git
+branch: master
+bucket_name: tinkerware.io
```
Comment (AAlvz) [inline ansible-s3-static-page-upload-role/defaults/main.yml]: missing end of line :(

File: ansible-s3-static-page-upload-role/vars/main.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+## These are the variables included in the defaults file
+# repo: git@github.com:Tinker-Ware/homepage.git
+# branch: master
+# bucket_name: tinkerware.io
```
Comment (AAlvz) [inline ansible-s3-static-page-upload-role/vars/main.yml]: missing end of line. 
Why are the lines commented? I don't think there's a need 🤔

File: main.yml [hunk 3]
```diff
@@ -1,55 +1,4 @@
...
-      file:
-        state: absent
-        path: homepage/
-
-    - name: Remove static-files folder
-      file:
-        state: absent
-        path: static-files/
+  roles:
+    - { role: ansible-s3-static-page-upload-role }
```
Comment (AAlvz) [inline main.yml]: missing end of line

File: variables.tf [hunk 4]
```diff
@@ -0,0 +1,9 @@
+variable "bucket_name" {
+  type = "string"
+  description = "Name for the bucket and the domain which will be used to access the website"
+}
+
+variable "create_www_bucket" {
+  type = bool
+  description = "Conditional to create or not the resources needed for the www subdomain"
+}
```
Comment (AAlvz) [inline variables.tf]: missing end of line

File: main.tf [hunk 5]
```diff
@@ -46,21 +48,23 @@ resource "aws_route53_record" "main_record" {
 }
 
 resource "aws_route53_record" "subdomain_record" {
+  count = "${ var.create_www_bucket ? 1 : 0 }"
```
Comment (AAlvz) [inline main.tf]: why are we counting?

**Review by AAlvz — CHANGES_REQUESTED** [review]
I just added requests for small changes and a couple of questions :)

File: main.tf (hunk 5 above)
Comment (estradag) [reply main.tf]: (Kind of) validation to create or not the resource if the bucket for the `www` subdomain is required

File: ansible-s3-static-page-upload-role/vars/main.yml (hunk 2 above)
Comment (estradag) [reply ansible-s3-static-page-upload-role/vars/main.yml]: @cracos suggestion, so there's no need to look inside `defaults` folder to see the variables set there.

Comment (AAlvz) [reply ansible-s3-static-page-upload-role/vars/main.yml]: the default variables exist for the cases when you don't want to change anything or look at the code and just "make this thing run" . That's why they should exist. To take the default ones if no other variables are defined and avoid failures.

File: main.tf (hunk 5 above)
Comment (AAlvz) [reply main.tf]: then should we change the name of that value? from `count` to something like `create_bucket` (True: 1, False: 0)

Comment (estradag) [reply main.tf]: `count` is a reserved word in Terraform, that's why the ternary operator only gives the values `1` and `0`, to create or not, respectively the bucket for that subdomain.

The variable `create_www_bucket` is the boolean used to trigger the ternary conditional.

Comment (AAlvz) [reply main.tf]: Ah true D:

File: ansible-s3-static-page-upload-role/vars/main.yml (hunk 2 above)
Comment (AAlvz) [reply ansible-s3-static-page-upload-role/vars/main.yml]: I was looking at the wrong file. Everything seems fine. Move on

**Review by AAlvz — APPROVED** [review]
LGTM

**Pull request merged**